pkg/controllers/state/metrics.go

```diff
@@ -0,0 +1,448 @@
+		},
+		nodeLabelNames(),
+	)
+
+	// Pod Metrics
+	podGaugeVec = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "karpenter",
+			Subsystem: "pods",
+			Name:      "state",
+			Help:      "Pod state is the current state of pods. This metric can be used several ways as it is labeled by the pod name, namespace, owner, node, provisioner name, zone, architecture, capacity type, instance type and pod phase.",
+		},
+		podLabelNames(),
+	)
+
+	// Provisioner Metrics
+	limitGaugeVec = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "karpenter",
+			Subsystem: "provisioner",
+			Name:      "limit",
+			Help:      "The Provisioner Limits are the limits specified on the provisioner that restrict the quantity of resources provisioned. Labeled by provisioner name and resource type.",
+		},
+		provisionerLabelNames(),
+	)
+	usageGaugeVec = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "karpenter",
+			Subsystem: "provisioner",
+			Name:      "usage",
+			Help:      "The Provisioner Usage is the amount of resources that have been provisioned by a particular provisioner. Labeled by provisioner name and resource type.",
+		},
+		provisionerLabelNames(),
+	)
+	usagePctGaugeVec = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "karpenter",
+			Subsystem: "provisioner",
+			Name:      "usage_pct",
+			Help:      "The Provisioner Usage Percentage is the percentage of each resource used based on the resources provisioned and the limits that have been configured in the range [0,100].  Labeled by provisioner name and resource type.",
+		},
+		provisionerLabelNames(),
+	)
+
+	prometheusMetrics = []*prometheus.GaugeVec{
+		allocatableGaugeVec,
+		podRequestsGaugeVec,
+		podLimitsGaugeVec,
+		daemonRequestsGaugeVec,
+		daemonLimitsGaugeVec,
+		overheadGaugeVec,
+		podGaugeVec,
+		limitGaugeVec,
+		usageGaugeVec,
+		usagePctGaugeVec,
+	}
+)
+
+func init() {
+	for _, m := range prometheusMetrics {
+		crmetrics.Registry.MustRegister(m)
+	}
+}
+
+func nodeLabelNames() []string {
+	return []string{
+		resourceType,
+		nodeName,
+		nodeProvisioner,
+		nodeZone,
+		nodeArchitecture,
+		nodeCapacityType,
+		nodeInstanceType,
+		nodePhase,
+	}
+}
+
+func podLabelNames() []string {
+	return []string{
+		podName,
+		podNameSpace,
+		ownerSelfLink,
+		podHostName,
+		podProvisioner,
+		podHostZone,
+		podHostArchitecture,
+		podHostCapacityType,
+		podHostInstanceType,
+		podPhase,
+	}
+}
+
+func provisionerLabelNames() []string {
+	return []string{
+		resourceType,
+		provisionerName,
+	}
+}
+
+func ResetMetrics() {
+	for _, vec := range prometheusMetrics {
+		vec.Reset()
+	}
+}
+
+// Gets the labels for a given pod based on current cluster state
+func (c *PodController) getPodLabels(ctx context.Context, pod *v1.Pod) prometheus.Labels {
```

Comment:
I'm not a fan of splitting up an implementation across source files unless there's a good reason.